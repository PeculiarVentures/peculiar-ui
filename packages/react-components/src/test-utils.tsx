import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
