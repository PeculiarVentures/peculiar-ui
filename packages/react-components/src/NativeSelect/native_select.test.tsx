import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { NativeSelect } from './index';

describe('<NativeSelect />', () => {
  const options: Partial<React.ComponentProps<typeof NativeSelect>>['options'] = [
    {
      label: 'test-1', value: 'test1',
    },
    {
      label: 'test-2', value: 'test3',
    },
  ];

  it('should render with default styles', () => {
    const {
      asFragment,
    } = render(
      <NativeSelect options={options} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', () => {
    const {
      asFragment,
    } = render(
      <NativeSelect options={options} className="my-class-name" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  describe('sizes', () => {
    const sizes: React.ComponentProps<typeof NativeSelect>['size'][] = [
      'small',
      'medium',
      'large',
    ];

    sizes.forEach((size) => {
      it(`size "${size}"`, () => {
        const {
          asFragment,
        } = render(
          <NativeSelect options={options} size={size} />,
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
