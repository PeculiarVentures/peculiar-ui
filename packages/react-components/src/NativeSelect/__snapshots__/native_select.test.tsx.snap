// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<NativeSelect /> > should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
}

.emotion-1 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 calc(var(--pv-size-base-2) + 24px) 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: block;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-weight: var(--pv-text-c1-weight);
  font-size: var(--pv-text-c1-size);
  line-height: var(--pv-text-c1-height);
  letter-spacing: var(--pv-text-c1-spacing);
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-1:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-1:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-1:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-1:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

.emotion-2 {
  position: absolute;
  right: 0px;
  top: calc(50% - 12px);
  pointer-events: none;
  margin: 0px var(--pv-size-base);
  color: var(--pv-color-gray-10);
}

<div
    class="my-class-name"
  >
    <div
      class="emotion-0"
    >
      <select
        class="emotion-1"
      >
        <option
          value="test1"
        >
          test-1
        </option>
        <option
          value="test3"
        >
          test-2
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="emotion-2"
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.39 14.512a.5.5 0 01-.78 0l-2.96-3.7A.5.5 0 019.04 10h5.92a.5.5 0 01.39.812l-2.96 3.7z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<NativeSelect /> > should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
}

.emotion-1 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 calc(var(--pv-size-base-2) + 24px) 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: block;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-weight: var(--pv-text-c1-weight);
  font-size: var(--pv-text-c1-size);
  line-height: var(--pv-text-c1-height);
  letter-spacing: var(--pv-text-c1-spacing);
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-1:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-1:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-1:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-1:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

.emotion-2 {
  position: absolute;
  right: 0px;
  top: calc(50% - 12px);
  pointer-events: none;
  margin: 0px var(--pv-size-base);
  color: var(--pv-color-gray-10);
}

<div>
    <div
      class="emotion-0"
    >
      <select
        class="emotion-1"
      >
        <option
          value="test1"
        >
          test-1
        </option>
        <option
          value="test3"
        >
          test-2
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="emotion-2"
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.39 14.512a.5.5 0 01-.78 0l-2.96-3.7A.5.5 0 019.04 10h5.92a.5.5 0 01.39.812l-2.96 3.7z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<NativeSelect /> > sizes > size "large" 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
}

.emotion-1 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 calc(var(--pv-size-base-2) + 24px) 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: block;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-weight: var(--pv-text-c1-weight);
  font-size: var(--pv-text-c1-size);
  line-height: var(--pv-text-c1-height);
  letter-spacing: var(--pv-text-c1-spacing);
  height: var(--pv-size-base-8);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-1:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-1:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-1:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-1:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

.emotion-2 {
  position: absolute;
  right: 0px;
  top: calc(50% - 12px);
  pointer-events: none;
  margin: 0px var(--pv-size-base);
  color: var(--pv-color-gray-10);
}

<div>
    <div
      class="emotion-0"
    >
      <select
        class="emotion-1"
      >
        <option
          value="test1"
        >
          test-1
        </option>
        <option
          value="test3"
        >
          test-2
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="emotion-2"
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.39 14.512a.5.5 0 01-.78 0l-2.96-3.7A.5.5 0 019.04 10h5.92a.5.5 0 01.39.812l-2.96 3.7z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<NativeSelect /> > sizes > size "medium" 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
}

.emotion-1 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 calc(var(--pv-size-base-2) + 24px) 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: block;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-weight: var(--pv-text-c1-weight);
  font-size: var(--pv-text-c1-size);
  line-height: var(--pv-text-c1-height);
  letter-spacing: var(--pv-text-c1-spacing);
  height: var(--pv-size-base-7);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-1:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-1:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-1:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-1:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

.emotion-2 {
  position: absolute;
  right: 0px;
  top: calc(50% - 12px);
  pointer-events: none;
  margin: 0px var(--pv-size-base);
  color: var(--pv-color-gray-10);
}

<div>
    <div
      class="emotion-0"
    >
      <select
        class="emotion-1"
      >
        <option
          value="test1"
        >
          test-1
        </option>
        <option
          value="test3"
        >
          test-2
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="emotion-2"
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.39 14.512a.5.5 0 01-.78 0l-2.96-3.7A.5.5 0 019.04 10h5.92a.5.5 0 01.39.812l-2.96 3.7z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<NativeSelect /> > sizes > size "small" 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
}

.emotion-1 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  width: 100%;
  border-radius: 4px;
  padding: 0 calc(var(--pv-size-base-2) + 24px) 0 var(--pv-size-base-2);
  background-color: var(--pv-color-gray-1);
  border-style: solid;
  border-width: 1px;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: block;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-weight: var(--pv-text-c1-weight);
  font-size: var(--pv-text-c1-size);
  line-height: var(--pv-text-c1-height);
  letter-spacing: var(--pv-text-c1-spacing);
  height: var(--pv-size-base-6);
  color: var(--pv-color-black);
  border-color: var(--pv-color-gray-8);
}

.emotion-1:hover {
  border-color: var(--pv-color-gray-10);
}

.emotion-1:disabled {
  cursor: not-allowed;
  border-color: var(--pv-color-gray-5);
  color: var(--pv-color-gray-7);
}

.emotion-1:not(:disabled)[aria-invalid] {
  border-color: var(--pv-color-wrong);
}

.emotion-1:not(:disabled):focus-visible {
  border-color: var(--pv-color-secondary);
}

.emotion-2 {
  position: absolute;
  right: 0px;
  top: calc(50% - 12px);
  pointer-events: none;
  margin: 0px var(--pv-size-base);
  color: var(--pv-color-gray-10);
}

<div>
    <div
      class="emotion-0"
    >
      <select
        class="emotion-1"
      >
        <option
          value="test1"
        >
          test-1
        </option>
        <option
          value="test3"
        >
          test-2
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="emotion-2"
        fill="none"
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.39 14.512a.5.5 0 01-.78 0l-2.96-3.7A.5.5 0 019.04 10h5.92a.5.5 0 01.39.812l-2.96 3.7z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
